### PR TITLE
[opt](join)Support MethodOneString to optimize hash join with a single string key

### PR DESCRIPTION
**Pull request by Mryange**
## Proposed changes

```
mysql [test]>SELECT count() from hits_10m WHERE SearchPhrase IN (SELECT SearchPhrase from hits_10m);
+----------+
| count(*) |
+----------+
| 10000000 |
+----------+
1 row in set (1.15 sec)

now
mysql [test]>SELECT count() from hits_10m WHERE SearchPhrase IN (SELECT SearchPhrase from hits_10m);
+----------+
| count(*) |
+----------+
| 10000000 |
+----------+
1 row in set (0.66 sec)

```

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

File: be/src/vec/common/hash_table/hash_map_context.h [hunk 1]
```diff
@@ -284,29 +285,59 @@ struct MethodStringNoCache : public MethodBase<TData> {
...
 
-    void init_serialized_keys(const ColumnRawPtrs& key_columns, size_t num_rows,
-                              const uint8_t* null_map = nullptr, bool is_join = false,
-                              bool is_build = false, uint32_t bucket_size = 0) override {
-        const IColumn& column = *key_columns[0];
-        const auto& column_string = assert_cast<const ColumnString&>(
-                column.is_nullable()
-                        ? assert_cast<const ColumnNullable&>(column).get_nested_column()
-                        : column);
+    void serialized_str_32(const ColumnString& column_string, std::vector<StringRef>& stored_keys) {
```
Comment (HappenLee) [inline be/src/vec/common/hash_table/hash_map_context.h:298]: use lambda to do the work

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by mrhhsg — APPROVED** [review]
lgtm

**Pull request merged**